HRMP Channel Setup Documentation (#34)
papermoonio/polkadot-ecosystem-docs-draft: docs/build-on-polkadot/xcm/hrmp-channels/para-to-system.md

---
title: Opening HRMP Channels with System Parachains
description: Learn how to open HRMP channels with Polkadot system parachains. Discover the process for establishing bi-directional communication using a single XCM message.
---

# Opening HRMP Channels with System Parachains

## Introduction

> While establishing HRMP channels between regular parachains involves a two-step request and acceptance procedure, opening channels with system parachains follows a more straightforward approach.

Reviewer: ```suggestion
While establishing Horizontal Relay-routed Message Passing (HRMP) channels between regular parachains involves a two-step request and acceptance procedure, opening channels with system parachains follows a more straightforward approach.
```
first time used on this page

System parachains are specialized chains that provide core functionality to the Polkadot network. Examples include Asset Hub for cross-chain asset transfers and Bridge Hub for connecting to external networks. Given their critical role, establishing communication channels with these system parachains has been optimized for efficiency and ease of use.

> Any parachain can establish a bi-directional channel with a system chain through a single operation, requiring just one XCM message from the parachain to the relay chain.

Reviewer: ```suggestion
Any parachain can establish a bidirectional channel with a system chain through a single operation, requiring just one XCM message from the parachain to the relay chain.
```

## Prerequisites

To successfully complete this process, you'll need to have the following in place:

- Access to a blockchain network consisting of:

>     1. A relay chain

Reviewer: ```suggestion
    - A relay chain
```

>     2. A parachain

Reviewer: ```suggestion
    - A parachain
```

>     3. An Asset Hub system chain

Reviewer: ```suggestion
    - An Asset Hub system chain
```

- A wallet containing enough funds to cover transaction fees on each of the participating chains

## Procedure for Establishing HRMP Channel

This guide demonstrates opening an HRMP channel between parachain 2500 and system chain Asset Hub (parachain 1000) on the Rococo Local relay chain.

### Step 1 - Fund Parachain Sovereign Account
<!-- This content will be moved to a new page because it is used in multiple places -->
The [sovereign account](https://github.com/polkadot-fellows/xcm-format/blob/10726875bd3016c5e528c85ed6e82415e4b847d7/README.md?plain=1#L50){target=_blank} for parachain 2500 on the relay chain must be funded so it can take care of any XCM transact fees.

> Use Polkadot.js Apps UI to connect to the relay chain and transfer funds from your account to the parachain 2500 sovereign account.

Reviewer: Let's link them to Apps UI on this first mention, please

        - Up/down chains use the prefix `0x70617261` (which decodes to `b"para"`)

        - Sibling chains use the prefix `0x7369626c` (which decodes to `b"sibl"`)

    2. Calculate the u32 scale encoded value of the parachain ID:
        - Parachain 2500 would be encoded as `c4090000`

    3. Combine the prefix and parachain ID encoding to form the full sovereign account address:

        The sovereign account of parachain 2500 in relay chain will be `0x70617261c4090000000000000000000000000000000000000000000000000000`
        and the SS58 format of this address is `5Ec4AhPSY2GEE4VoHUVheqv5wwq2C1HMKa7c9fVJ1WKivX1Y`
    
    To perform this conversion, you can also use the **"Para ID" to Address** section in [Substrate Utilities](https://www.shawntabrizi.com/substrate-js-utilities/).

### Step 2 - Create Establish Channel with System Extrinsic

1. In Polkadot.js Apps, connect to the relay chain, navigate to the **Developer** dropdown and select the **Extrinsics** option

    ![](/polkadot-ecosystem-docs-draft/images/build-on-polkadot/hrmp-channels/para-to-para/hrmp-para-to-para-1.webp)

2. Construct an `establish_channel_with_system` extrinsic call

>     1. Select the **hrmp** pallet

Reviewer: ```suggestion
    1. Select the **`hrmp`** pallet
```
